R/process_lookup_homelessness.R

```diff
@@ -0,0 +1,114 @@
+# It would probably be simplest to create a lookup from the Homelessness data of one row per CHI +
+#   start dates of all applications. This can then be matched to episode and individual to easily create the hl1_in_fy flag(s)
+# (if someone is in the lookup -> 1, otherwise -> 0). For the episode file, it is slightly more complicated to create the other flags,
+# either matching on all the data by CHI and then doing some comparison of dates (as in SPSS),
+# or using some of the fancy new joins (dplyr 1.1.0 joins) and doing it that way.
+#
+#
+
+
+# data <- slfhelper::read_slf_episode("1718", c("anon_chi", "record_keydate1", "record_keydate2", "recid"))
+
+# year <- "1718"
+
+
+#' create homelessness lookup
+#' @description reads in homelessness extract and selects CHIs and flags them as homeless in FY
+#'
+#' @param year The year to process, in FY format.
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @export
+#' @family process extracts
+#'
+create_homelessness_lookup <- function(year) {
+  homelessness_lookup <- read_file(get_source_extract_path(year, "Homelessness")) %>%
+    dplyr::distinct(chi, record_keydate1, record_keydate2) %>%
+    tidyr::drop_na(chi) %>%
+    dplyr::mutate(hl1_in_fy = 1L)
+
+  return(homelessness_lookup)
+}
+
+
+#' add homelessness flag episode
+#' @description add homelessness in FY flag to episode/individual file
+#'
+#' @param data The extract to process
+#' @param year The year to process, in FY format.
+#'
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @export
+#'
+
+add_homelessness_flag <- function(data, year) {
+  lookup <- create_homelessness_lookup(year) %>%
+    slfhelper::get_anon_chi()
+
+  ## need to decide which recids this relates to
+  data <- data %>%
+    dplyr::left_join(
+      lookup %>%
+        dplyr::distinct(anon_chi, hl1_in_fy),
+      by = "anon_chi", relationship = "many-to-one"
+    ) %>%
+    dplyr::mutate(hl1_in_fy = tidyr::replace_na(hl1_in_fy, 0L))
+
+  return(data)
+}
+
+
+#' add homelessness date flags episode
+#' @description flags episodes with homelessness applications in 6 months before, 6 months after, or during episode
+#'
+#' @param data The extract to process
+#' @param year The year to process, in FY format.
+#'
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @export
+#'
+add_homelessness_date_flags_episode <- function(data, year) {
+  lookup <- create_homelessness_lookup(year) %>%
+    slfhelper::get_anon_chi() %>% # TO DO - change back to get_chi
```

Comment:
As per the previous comment.
```suggestion
add_homelessness_date_flags_episode <- function(data, year, lookup = create_homelessness_lookup(year)) {
  lookup <- lookup %>%
```